libnemo-private/nemo-file.c

```diff
@@ -4036,6 +4065,28 @@ nemo_file_peek_display_name_collation_key (NemoFile *file)
 	return res;
 }
 
+const char *
+nemo_file_peek_extension_name (NemoFile *file)
+{
+    if (file == NULL) {
+        return NULL;
+    }
+
+    if (nemo_file_is_broken_symbolic_link (file)) {
+        return g_strdup (_("link (broken)"));
+	}
+	
+	char *str, *token, *result;
+	str = g_strdup (eel_ref_str_peek(file->details->name));//strdup()
+	
+	while ((token = strsep(&str, "."))) {result = g_strdup(token);}
+	if (result == NULL) {result = "";}
+	g_free(str);
+	g_free(token);
+	
+	return result;
+}
```

Comment:
There are a few problems here:

Remember you're not returning a new string (`const char*`), since you're 'peeking' at the value:
- link broken string should just be a static string (no g_strdup).
- The remaining code either returns a new string, or a static empty one, and leaks memory (you never free `token` while running the `strsep` loop, and if there are multiple separators the intermediate ones leak).

I would suggest using `g_strrstr` as it works from the end of the string and returns a pointer or `NULL`, and no new strings need to be created:
```
    const char *str, *ptr;
    str = eel_ref_str_peek (file->details->name);

    if ((ptr = g_strrstr (str, "."))) {
        return ptr + 1;
    }

    return "";
```
We usually try to stick with glib functions, as they're generally more convenient (and often safer).